### DISCUSSION TITLE
files di configurazione di HA nella cartella sbagliata?

### DISCUSSION THREAD

**Discussion by calcioscacchi** [indomus/forum#800] [HUB: Home Assistant]
SAlve, ho seguito la seguente guida per l'installazione di HA Core con docker:

https://indomus.it/guide/come-installare-e-configurare-home-assistant-core-con-docker-su-un-raspberry-pi-gia-in-uso/

Vi anticipo già che sono sicuro di non aver seguito invece questa guida: https://indomus.it/guide/come-installare-e-configurare-home-assistant-su-un-raspberry-pi-gia-in-uso/

E quindi mi aspetto di trovare il file configuration.yaml sulla directory /home/pi/homeassistant/.
Tuttavia, con mia sorpresa, quella cartella è vuota e mi trovo tutto installato sulla directory /home/homeassistant/.homeassistant/.

Ora, l'intuito mi dice che se segui una guida e il risultato è diverso da quello che ti aspettavi, NON VA BENE e quindi se trascuro il fatto potrei avere dei problemi. Vorrei anche un vostro parere in merito, prima di avventurarmi oltre nel processo  di configurazione. 
Grazie.

**Comment by snakuzzo** [comment]
A prescindere dal personale punto di vista su come va "interpretata" una guida...che è una guida e non un manuale di istruzioni da seguire con copia incolla senza comprendere bene cosa si sta andando a fare, probabilmente nel tuo caso hai mappato male il volume relativo alla directory `/config`

Se hai passato i valori come da guida...

`-v /home/pi/homeassistant:/config`

...la directory `/config` del container di HA (che è quella che va ad usare HA) viene mappata sulla directory dell'host `/home/pi/homeassistant`.

Se così non è, vuol dire che hai sbagliato qualcosa nel mapping dei volumi

**Reply by calcioscacchi** [reply]
Ciao Diego, 

ho capito l'errore... proprio per documentarmi meglio sui vari componenti della guida, devo aver preso il contenuto del file docker-compose.yaml dalla guida che tratta l'argomento: https://indomus.it/formazione/gestire-applicazioni-docker-tramite-docker-compose/

Mi ricordo bene che mi ero accorto dell'errore e ho corretto il file rilanciando il comando. Ok, frittata fatta... ma per risolvere, mi conviene rifare l'installaizone di tutto quando da capo o basta rimuovere Home assistant e riprendere la procedura da li, lasciando installato il docker?

**Reply by snakuzzo** [reply]
assolutamente no.
Ti basta ricreare il container passandogli la directory dove hai i file...oppure sposti i file sotto la directory che erroneamente (per modo di dire) hai configurato sul container...a te la scelta

**Reply by calcioscacchi** [reply]
perdona il ritardo nella risposta. 
Interpreto con "sposta tutto il contenuto della cartella /home/homeassistant/.homeassistant/ in /home/pi/homeassistant/."

Provo a vi faccio sapere.

**Reply by calcioscacchi** [reply]
allora, ho spostato tutto e quando sono entrato in dashboard tramite il mio ip (indirizzoip:8123) mi ha richiesto la configurazione iniziale, spero sia normale altrimenti nel dubbio reinstallo da capo tutto.

Inoltre, giocando un po' con il comando sudo mv, ma provando a lanciare l'aggiornamento di home assistant, esce questo errore ,non so se è dovuto al fatto di aver sbagliato percorso. Parla di un container che utilizza il container di home assistant, quindi mi fa pensare che potrebbe essere.

```
pi@raspberrypi:~ $ docker-compose pull home-assistant
Pulling home-assistant ... done
pi@raspberrypi:~ $ docker-compose up --build -d home-assistant
Creating home-assistant ... error

ERROR: for home-assistant  Cannot create container for service home-assistant: Conflict. The container name "/home-assistant" is already in use by container "5825a9039b86630a1fa967ea2afc1fb221e055d36839beb3635816204e71ce1d". You have to remove (or rename) that container to be able to reuse that name.

ERROR: for home-assistant  Cannot create container for service home-assistant: Conflict. The container name "/home-assistant" is already in use by container "5825a9039b86630a1fa967ea2afc1fb221e055d36839beb3635816204e71ce1d". You have to remove (or rename) that container to be able to reuse that name.
ERROR: Encountered errors while bringing up the project.
```

**Reply by snakuzzo** [reply]
Se ti chiede la configurazione iniziale hai di nuovo sbagliato a montare la directory di configurazione correttamente.
Inoltre per ricreare il container non serve `--build`. Una volta fatta la `pull` ti basta ridare il comando

`docker-compose up -d home-assistant`

**Reply by calcioscacchi** [reply]
ok, nel dubbio reinstallo tutto, se qualcosa non va niutile pensare che si sistemi da sola ;)

**Reply by snakuzzo** [reply]
Sì ma nemmeno reinstallare tutto è la soluzione...prova a capire meglio il discorso del mount dei volumi...insisti :)

**Reply by calcioscacchi** [reply]
vero. Ma temo che, essendo andato un po' avanti con la configurazione, lascerei qualcosa di incompleto/compromettente. Posso sempre fare entrambe le cose XD
